apps/web/src/components/data-table/monitor/data-table-row-actions.tsx

```diff
@@ -28,22 +29,47 @@ import {
 
 import { LoadingAnimation } from "@/components/loading-animation";
 import type { RegionChecker } from "@/components/ping-response-analysis/utils";
-import { toastAction } from "@/lib/toast";
+import { toast, toastAction } from "@/lib/toast";
 import { api } from "@/trpc/client";
-
+import { cn } from "@/lib/utils";
 interface DataTableRowActionsProps<TData> {
   row: Row<TData>;
 }
 
+const alerts = {
+  clone: {
+    action: {
+      variant: "default",
+      text: "Clone",
+    },
+    dialog: {
+      title: "Are you absolutely sure?",
+      description: "This action cannot be undone. This will clone the monitor.",
+    },
+  },
+  delete: {
+    action: {
+      text: "Delete",
+      variant: "destructive",
+    },
+    dialog: {
+      title: "Are you absolutely sure?",
+      description:
+        "This action cannot be undone. This will permanently delete the monitor.",
+    },
+  },
+} as const;
+
 export function DataTableRowActions<TData>({
   row,
 }: DataTableRowActionsProps<TData>) {
-  const { monitor } = z
-    .object({ monitor: selectMonitorSchema })
+  const { monitor, isLimitReached } = z
+    .object({ monitor: selectMonitorSchema, isLimitReached: z.boolean() })
     .parse(row.original);
   const router = useRouter();
-  const [alertOpen, setAlertOpen] = React.useState(false);
-  const [isPending, startTransition] = React.useTransition();
+  const [alertOpen, setAlertOpen] = useState(false);
+  const [alertType, setAlertType] = useState<"delete" | "clone">("delete");
```

Comment:
Let's use a single state.

```suggestion
  const [alertType, setAlertType] = useState<"delete" | "clone"| undefined>();
  const alertOpen = useMemo(() => !alertType, [alertType])
```

If the type is `undefined`, the open value will be `false`.

That will reduce the redundant state an avoid using `setAlertOpen`.

To reset, we can use `setAlertType(undefined)`.

Reply:
Hey @mxkaske ,
But how will me manage `onOpenChnage`  in `<AlertDialog open={alertOpen} onOpenChange={(value) => setAlertOpen(value)}>`

Reply:
We don't even need  `const alertOpen = useMemo(() => !alertType, [alertType])` we can use `!!alertType` But again how will we manage `onOpenChange`?